### PR TITLE
handle dev and main branch in get_target_branch.jl

### PR DESCRIPTION
**Pull request by SimeonEhrig**

### CODE REVIEW

File: .ci/integTestGen/src/get_target_branch.jl [hunk 1]
```diff
@@ -20,10 +20,16 @@ function get_target_branch()::AbstractString
         error("Environment variable CI_COMMIT_REF_NAME is not set.")
     end
 
+    if ENV["CI_COMMIT_REF_NAME"] == "main"
+        # run on main branch
+        return "main"
+    end
```
Comment (AntonReinhard) [inline .ci/integTestGen/src/get_target_branch.jl:26]: Does this trigger when running *on* the main branch or when running on a PR that is *targeting* the `main` branch?
At least the comment sounds like it's the first one, but it should be the second.

**Review by AntonReinhard — COMMENTED** [review]
This should hopefully fix the problems. I also just realized that there will still be branches on the main repository (even without the release-* branches), for example the compat-helper branches. In which case those will also target dev and should be built with dev dependencies, which they will now do :+1:

File: .ci/integTestGen/src/get_target_branch.jl (hunk 1 above)
Comment (SimeonEhrig) [reply .ci/integTestGen/src/get_target_branch.jl:26]: When it runs on the branch main.

From the doc: `CI_COMMIT_REF_NAME` : `The branch or tag name for which project is built.`

https://docs.gitlab.com/ee/ci/variables/predefined_variables.html

Also GitLab, the mirror does not open an Merge Request. It simply creates a new branch, push the code on it and returns the result from testing the last commit.

Comment (AntonReinhard) [reply .ci/integTestGen/src/get_target_branch.jl:26]: Okay I looked at the entire file now and I think I understand what's happening.

**Review by AntonReinhard — APPROVED** [review]

**Pull request merged**